compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/common/Util.kt

```diff
@@ -119,6 +119,20 @@ fun MethodNode.removeEmptyCatchBlocks() {
 
 fun MethodNode.removeUnusedLocalVariables() {
     val used = BooleanArray(maxLocals) { false }
+
+    // Arguments are always used whether or not they are in the local variable table
+    // or used by instructions.
+    var argumentIndex = 0
+    val isStatic = (access and Opcodes.ACC_STATIC) != 0
+    if (isStatic) {
+        used[argumentIndex++] = true
```

Comment:
It looks like you meant the opposite: method uses argument slot 0 if it is NOT static.

Reply:
Whoops, yes, that was a last minute change.

Reply:
Thanks for catching this Dmitry.

Just following up: I tried to create a test case for this typo, however, that is not easy. I haven't been able to create a reproduction without using suspend functions. For suspend functions, accidentally not marking the receiver means that we end up marking all the arguments except for the last one. That one is the continuation and that is always used by instructions and is therefore kept because it is used by instructions. Therefore, this actually works with or without correct marking of the receiver local.